utils/generate_dotnet_channels.js

```diff
@@ -126,100 +162,359 @@ function objectType(props, indent, onlyOptional = false) {
+  initializer.push(`{`);
+  initializer.push(`    internal class ${initializerName}${superName ? ' : ' + superName + 'Initializer' : ''}`);
+  initializer.push(init.ts);
+  initializer.push(`}`);
+  initializer.push(``);
+  writeFile(`${initializerName}.cs`, initializer.join('\n'), protocolDir);
 }
 
-/**
- * @param {string} name
- * @returns {string}
- */
-function toTitleCase(name) {
-  return name.charAt(0).toUpperCase() + name.substring(1);
+function writeFile(file, content, folder) {
+  fs.writeFileSync(path.join(folder, file), content, 'utf8');
 }
 
-function processCustomType(type, optional) {
+function processCustomType(type, optional, parentName) {
   if (type.properties.name
-      && type.properties.value
-      && inlineType(type.properties.name).ts === 'string'
-      && inlineType(type.properties.value).ts === 'string') {
+    && type.properties.value
+    && inlineType(type.properties.name).ts === 'string'
+    && inlineType(type.properties.value).ts === 'string')
     return { ts: 'HeaderEntry', scheme: 'tObject()', optional };
-  }
+
   if (type.properties.width
-      && type.properties.height
-      && inlineType(type.properties.width).ts === 'int'
-      && inlineType(type.properties.height).ts === 'int') {
+    && type.properties.height
+    && inlineType(type.properties.width).ts === 'int'
+    && inlineType(type.properties.height).ts === 'int')
     return { ts: 'ViewportSize', scheme: 'tObject()', optional };
-  }
+
   if (type.properties.url
     && type.properties.lineNumber
     && inlineType(type.properties.url).ts === 'string'
-    && inlineType(type.properties.lineNumber).ts === 'int') {
+    && inlineType(type.properties.lineNumber).ts === 'int')
     return { ts: 'ConsoleMessageLocation', scheme: 'tObject()', optional };
-  }
+
   if (type.properties.name
     && type.properties.descriptor
-    && inlineType(type.properties.name).ts === 'string') {
+    && inlineType(type.properties.name).ts === 'string')
     return { ts: 'DeviceDescriptorEntry', scheme: 'tObject()', optional };
+
+  if (parentName === 'newDocument')
+    return { ts: 'NavigateDocument', scheme: 'tObject()', optional };
+
+  if (parentName === 'getPropertyList')
+    return { ts: `JSElementProperty`, scheme: `tObject()`, optional };
+
+  // TODO: this needs to be fixed
+  if (['startJSCoverage', 'stopJSCoverage', 'startCSSCoverage', 'stopCSSCoverage'].includes(parentName))
+    return { ts: `JsonElement`, scheme: `tObject()`, optional };
+
+  if ('setInputFiles')
+    return { ts: `FilePayload`, scheme: `tObject()`, optional };
+
+  if (type.properties
+    && type.properties.value
+    && type.properties.label
+    && type.properties.index)
+    return { ts: `SelectOptionValue`, scheme: `tObject()`, optional };
+}
+
+function cleanUpName(name) {
+  switch (name) {
+    case "event":
+    case "params":
+      return `@${name}`;
+    default:
+      return name;
   }
-}
\ No newline at end of file
+}
+
+function buildParameterMap(parameters) {
```

Comment:
Consider annotating function types in this file as it keeps growing and becomes harder  to understand/validate